pkg/client/unversioned/discovery_client.go

```diff
@@ -71,42 +69,29 @@ func apiVersionsToAPIGroup(apiVersions *unversioned.APIVersions) (apiGroup unver
 	return
 }
 
-func (d *DiscoveryClient) get(url string) (resp *http.Response, err error) {
-	req, err := http.NewRequest("GET", url, nil)
-	if err != nil {
-		return nil, err
-	}
-	return d.httpClient.Do(req)
-}
-
 // ServerGroups returns the supported groups, with information like supported versions and the
 // preferred version.
 func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList, err error) {
 	// Get the groupVersions exposed at /api
-	url := d.baseURL
-	url.Path = "/api"
-	resp, err := d.get(url.String())
-	if err != nil {
-		return nil, err
+	v := &unversioned.APIVersions{}
+	err = d.Get().AbsPath("/api").Do().Into(v)
+	apiGroup := unversioned.APIGroup{}
+	if err == nil {
+		apiGroup = apiVersionsToAPIGroup(v)
 	}
-	var v unversioned.APIVersions
-	defer resp.Body.Close()
-	err = json.NewDecoder(resp.Body).Decode(&v)
-	if err != nil {
-		return nil, fmt.Errorf("unexpected error: %v", err)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
 	}
-	apiGroup := apiVersionsToAPIGroup(&v)
 
 	// Get the groupVersions exposed at /apis
-	url.Path = "/apis"
-	resp2, err := d.get(url.String())
-	if err != nil {
-		return nil, err
-	}
-	defer resp2.Body.Close()
 	apiGroupList = &unversioned.APIGroupList{}
-	if err = json.NewDecoder(resp2.Body).Decode(&apiGroupList); err != nil {
-		return nil, fmt.Errorf("unexpected error: %v", err)
+	err = d.Get().AbsPath("/apis").Do().Into(apiGroupList)
+	if err != nil {
+		if !errors.IsNotFound(err) {
```

Comment:
I'm not familiar with the `secured server`. Could you point me to the related code? Thanks.